tests/tools/test_processes.py

```diff
@@ -2,56 +2,547 @@
+
+        execute_calls = [(0, ps_output, ""), (0, status_output, "")]
+
+        mocker.patch.object(processes, "execute_command", side_effect=execute_calls)
+
+        result = await mcp.call_tool("get_process_info", {"pid": 1, "host": "remote.example.com"})
+        text = _get_text_from_mcp_result(result)
+
+        assert "Process Information for PID 1" in text
+        assert "systemd" in text.lower() or "init" in text.lower()
+        assert "Detailed Status" in text
+
+    async def test_get_process_info_with_host_no_proc_status(self, mocker):
+        """Test get_process_info on remote host without /proc status."""
+        ps_output = "  PID USER     STAT  %CPU %MEM    VSZ   RSS     ELAPSED COMMAND         COMMAND\n    1 root     Ss     0.0  0.1 169564 13808    10:23:45 systemd         /sbin/init\n"
+
+        execute_calls = [(0, ps_output, ""), (1, "", "cat: /proc/1/status: Permission denied")]
+
+        mocker.patch.object(processes, "execute_command", side_effect=execute_calls)
+
+        result = await mcp.call_tool("get_process_info", {"pid": 1, "host": "remote.example.com"})
+        text = _get_text_from_mcp_result(result)
+
+        assert "Process Information for PID 1" in text
+        assert "Detailed Status" not in text
+
+    async def test_get_process_info_with_host_nonexistent(self, mocker):
+        """Test get_process_info on remote host with nonexistent PID."""
+        mocker.patch.object(processes, "execute_command", return_value=(1, "", "Error"))
+
+        result = await mcp.call_tool("get_process_info", {"pid": 99999, "host": "remote.example.com"})
+        text = _get_text_from_mcp_result(result)
+
+        assert "does not exist on remote host" in text
+
+    async def test_get_process_info_with_host_empty_output(self, mocker):
+        """Test get_process_info on remote host with empty output."""
+        mocker.patch.object(processes, "execute_command", return_value=(0, "", ""))
+
+        result = await mcp.call_tool("get_process_info", {"pid": 1, "host": "remote.example.com"})
+        text = _get_text_from_mcp_result(result)
+
+        assert "does not exist on remote host" in text
+
+    async def test_get_process_info_with_host_invalid_pid(self, mocker):
+        """Test get_process_info on remote host with invalid PID."""
+        result = await mcp.call_tool("get_process_info", {"pid": -5, "host": "remote.example.com"})
+        text = _get_text_from_mcp_result(result)
+        assert "Error: PID must be at least 1" in text
+
+    async def test_list_processes(self):
+        """Test calling list_processes through mcp.call_tool()."""
+        result = await mcp.call_tool("list_processes", {})
+
+        # mcp.call_tool returns a tuple of (list[TextContent], dict)
+        assert isinstance(result, tuple)
+        assert len(result) == 2
```

Comment:
I'm not sure these statements add much value. It would be better to do this once rather than in every test. The interface in FastMCP is totally different, which means all of these tests need to be rewritten. Reducing the number of times that needs to be done would be ideal.